src/BytesLib.sol

```diff
@@ -12,13 +12,17 @@ pragma solidity ^0.8.25;
 library FlatBytesLib {
     using FlatBytesLib for *;
 
+    error InvalidDataLength();
+
+    uint256 constant maxDataLength = 32;
+
     /*//////////////////////////////////////////////////////////////////////////
                                     DATA STRUCTURES
     //////////////////////////////////////////////////////////////////////////*/
 
     // Data structure to store bytes in consecutive slots using an array
     struct Data {
-        bytes32[10] slot1;
+        bytes32[maxDataLength] slot1;
```

Comment:
Is this right? Now the array length is 32 and the calculations below are still for 10 x 32